### DISCUSSION TITLE
Seed

### DISCUSSION THREAD

**Discussion by perceptualrobots** [cyberbotics/wrestling#69] [Q&A] [answered]
I see from Supervisor.simulationReset "Resets random seeds". I presume this means that the initial values will be the same each time.

For testing purposes, I would like the same values each time. Is there a way I can run the same seed each time even though calling simulationReset?

**Comment by omichel** [comment, marked as answer]
On each run or simulation reset, Webots will use the same random seed values. However, the wrestling simulation is asynchronous and thus, the results are not reproducible as they depend on the system scheduler. In order to have reproducible simulations, you should set each robot to "synchronization TRUE".

**Comment by perceptualrobots** [comment]
Hi, 

I am not finding that results are completely repeatable.

That is, everything is repeatable until the two robots interact. Do you know why that might be?

The blue robot is the default non-moving.

For the red robot, I am controlling the legs from a hierarchical control system setting the position of three leg motors for each leg. For the first 16 seconds, as red walks towards blue all sensor readings and motor values are exactly the same on each run. However, once red reaches blue and they interact there are differences in what happens on each run.

Both have "synchronization TRUE".

**Reply by omichel** [reply]
Yes, unfortunately, simulations are not always 100% reproducible. You may want to investigate it why by inspecting the source code of Webots and trying to find out when and why the results are diverging.
